src/Fixer/FunctionNotation/VoidReturnFixer.php

```diff
@@ -0,0 +1,251 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        // These cause syntax errors.
+        static $blacklistFuncNames = [
+            [T_STRING, '__construct'],
+            [T_STRING, '__destruct'],
+            [T_STRING, '__clone'],
+        ];
+
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $funcName = $tokens->getNextMeaningfulToken($index);
+            if ($tokens[$funcName]->equalsAny($blacklistFuncNames, false)) {
+                continue;
+            }
+
+            $startIndex = $tokens->getNextTokenOfKind($index, ['{', ';']);
+
+            if ($this->hasReturnTypeHint($tokens, $startIndex)) {
+                continue;
+            }
+
+            if (!$tokens[$startIndex]->equals('{')) {
+                // No function body defined, fallback to PHPDoc.
+                if ($this->hasVoidReturnAnnotation($tokens, $index)) {
+                    $this->fixFunctionDefinition($tokens, $startIndex);
+                }
+
+                continue;
+            }
+
+            if ($this->hasReturnAnnotation($tokens, $index)) {
+                continue;
+            }
+
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $startIndex);
+
+            if ($this->hasVoidReturn($tokens, $startIndex, $endIndex)) {
+                $this->fixFunctionDefinition($tokens, $startIndex);
+            }
+        }
+    }
+
+    /**
+     * Determine if there is a non-void return annotation in the function's PHPDoc comment.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the function token
+     *
+     * @return bool
+     */
+    private function hasReturnAnnotation(Tokens $tokens, $index)
+    {
+        foreach ($this->findReturnAnnotations($tokens, $index) as $return) {
+            $types = $return->getTypes();
+            if (count($types) > 1 || 'void' !== $types[0]) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Determine if there is a void return annotation in the function's PHPDoc comment.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the function token
+     *
+     * @return bool
+     */
+    private function hasVoidReturnAnnotation(Tokens $tokens, $index)
+    {
+        foreach ($this->findReturnAnnotations($tokens, $index) as $return) {
+            $types = $return->getTypes();
+            if (1 === count($types) && 'void' === $types[0]) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Determine the function already has a return type hint.
```

Comment:
All PHPDoc descriptions in this class use _Determine if_ so I would use it here as well... or update all the others :)